safe/gis/tools.py

```diff
@@ -249,10 +250,12 @@ def load_layer(full_layer_uri_string, name=None, provider=None):
         keywords = read_iso19115_metadata(layer.source())
         if 'layer_purpose' in keywords:
             layer_purpose = keywords['layer_purpose']
-    except NoKeywordsFoundError:
-        pass
+    except (NoKeywordsFoundError, MetadataReadError):
+        keywords = {}
 
-    monkey_patch_keywords(layer)
+    if not keywords.get('inasafe_fields'):
+        keywords['inasafe_fields'] = {}
+    layer.keywords = keywords
```

Comment:
I can't remember exactly the reason. But because this monkey patching is a kind of hack in the QgsMapLayer class (not part of the official QgsMapLayer API), I think I wanted to centralize this hack in a single function. But do as you want

Reply:
sure we can centralize it using this monkey patching method, and remove the upper code instead because it's duplicated. let me update it. thanks for the review!